modules/hgraph.R

```diff
@@ -0,0 +1,201 @@
+library(dplyr)
+library(reshape2)
+library(ggplot2)
+library(stringr)
+
+plainToPlot <- function(myPlainText) {
+  # Convert NAs to blank spaces
+  myPlainText[is.na(myPlainText)] <- ""
+
+  # Change the object type
+  myPlainText <- as.character(myPlainText)
+
+  # Remove unnecessary spaces
+  myPlainText <- gsub(" \\\\n ", "\\\\n", myPlainText)
+  myPlainText <- gsub(" \\\\n", "\\\\n", myPlainText)
+  myPlainText <- gsub("\\\\n ", "\\\\n", myPlainText)
+
+  temp <- NULL
+
+  for (j in 1:length(myPlainText)) {
+    myPlainTextSub <- myPlainText[j]
+    # Get Number of line breaks
+    lineBreaks <- str_count(myPlainTextSub, pattern = "\\\\n")
+    if (lineBreaks == 0) {
+      finalExpression <- myPlainTextSub
+    } else if (lineBreaks == 1) {
+      separateWords <- strsplit(myPlainTextSub, "\\\\n")[[1]]
+      finalExpression <- paste0("atop(textstyle(", separateWords[1], "),textstyle(", separateWords[2], "))")
+    } else if (lineBreaks > 1) {
+      separateWords <- strsplit(myPlainTextSub, "\\\\n")[[1]]
+      finalExpression <- paste0("atop(textstyle(", separateWords[1], "),textstyle(", separateWords[2], "))")
+      # Initiate loop
+      for (i in 2:lineBreaks) {
+        finalExpression <- paste0("atop(textstyle(", finalExpression, "),textstyle(", separateWords[i + 1], "))")
+      }
+    }
+    temp <- c(temp, finalExpression)
+  }
+  temp <- gsub(" ", "~", temp)
+  return(temp)
+}
+
+
+options(scipen = 999)
+
+theme_nothing <- function(base_size = 12, base_family = "Helvetica") {
+  theme_bw(base_size = base_size, base_family = base_family) %+replace%
+    theme(
+      rect             = element_blank(),
+      line             = element_blank(),
+      text             = element_blank(),
+      legend.position  = "none"
+    )
+}
+
+radians <- function(x, y, xend, yend, halfWid, halfHgt) {
+  theta <- sign(yend - y) * (abs(xend - x) < .0001) * pi / 2
+  indx <- (xend - x > 0.00001)
+  theta[indx] <- atan((yend[indx] - y[indx]) * halfWid / (xend[indx] - x[indx]) / halfHgt)
+  indx <- ((xend - x) < -0.00001 & (yend - y) > .00001)
+  theta[indx] <- pi / 2 + atan(-(xend[indx] - x[indx]) * halfHgt / (yend[indx] - y[indx]) / halfWid)
+  indx <- ((xend - x) < -.00001 & (yend - y) < -.00001)
+  theta[indx] <- atan((yend[indx] - y[indx]) / halfHgt / (xend[indx] - x[indx]) * halfWid) + pi
+  indx <- ((xend - x) < -.00001 & abs(yend - y) < .00001)
+  theta[indx] <- pi
+  return(theta)
+}
+
+latestCall <- "" # global variable needed to store input args for a call to hGraph function
+
+
+hGraph <- function(nHypotheses = 4, # Number of hypotheses
```

Comment:
We also need to replace this internal `hGraph` function with the `gMCPmini::hGraph()` or `gsDesign::hGraph()`, a few argument names have been changed...let's update this in another branch.

Reply:
Oh sure - just to clarify, I didn't touch any of the content or logic in this PR - only refactored the existing code using a more manageable ui/server file structure.